lib/koala/api/batch_operation.rb

```diff
@@ -28,9 +28,15 @@ def initialize(options = {})
           raise AuthenticationError.new(nil, nil, "Batch operations require an access token, none provided.") unless @access_token
         end
 
-        def to_batch_params(main_access_token)
+        def to_batch_params(main_access_token, main_app_secret)
           # set up the arguments
-          args_string = Koala.http_service.encode_params(@access_token == main_access_token ? @args : @args.merge(:access_token => @access_token))
+          if @access_token != main_access_token
+            @args[:access_token] = @access_token
+            if main_app_secret
+              @args[:appsecret_proof] = OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new("sha256"), main_app_secret, @access_token)
```

Comment:
To the best of my understanding, it's only required if the access_token for a given batch element is different from the access_token used for the overall batch request.